hazelcast/src/main/java/com/hazelcast/internal/util/collection/ImmutableInflatableSet.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.util.collection;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Predicate;
+
+import com.hazelcast.nio.serialization.SerializableByConvention;
+
+/**
+ * An immutable {@link InflatableSet}.
+ *
+ * @param <T> the type of elements maintained by this set
+ */
+@SerializableByConvention
+public final class ImmutableInflatableSet<T> extends InflatableSet<T> {
+
+    private static final long serialVersionUID = 1L;
+
+    private ImmutableInflatableSet(List<T> compactList) {
+        super(compactList);
+    }
+
+    public static <T> ImmutableBuilder<T> newImmutableBuilder(int initialCapacity) {
```

Comment:
I already have `newBuilder(int)` in the superclass. Maybe `newImmutableSetBuilder`? To avoid confusion I can rename class `ImmutableBuider->ImmutableSetBuilder`

Reply:
ah, I see. `newImmutableSetBuilder` sounds good to me.